controllers/event_watcher_registry.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package controllers
+
+import (
+	"context"
+
+	gocache "github.com/patrickmn/go-cache"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// eventWatcherMeta - metadata for managing an event watcher goroutine
+type eventWatcherMeta struct {
+	// Cancel will close the watcher's context (and stop the watcher goroutine)
+	Cancel context.CancelFunc
+	// StoppedCh lets the watcher goroutine signal the caller that it has
+	// stopped (and removed itself from the registry)
+	StoppedCh chan struct{}
+	// LastGeneration is the generation of the VaultStaticSecret resource, used
+	// to detect if the event watcher needs to be recreated
+	LastGeneration int64
+	// LastClientID - vault client ID for the last successful connection, used
+	// to detect if the Vault client has changed since the event watcher started
+	LastClientID string
+}
+
+type EventWatcherRegistry struct {
```

Comment:
nit: would be good to add some docs for this exported struct.

Reply:
Sure, and I un-exported it in 2b28b51 since it wasn't being used outside the package.